src/backend/distributed/executor/multi_server_executor.c

```diff
@@ -111,7 +113,8 @@ JobExecutorType(DistributedPlan *distributedPlan)
 			}
 			if (HasReplicatedDistributedTable(distributedPlan->relationIdList))
 			{
-				return MULTI_EXECUTOR_TASK_TRACKER;
+				ereport(ERROR, (errmsg(
+									"repartitioning with shard_replication_factor > 1 is not supported")));
```

Comment:
I believe we can actually support it via adaptive executor, but not do the same level of failure handling.

Reply:
Does it make sense to support it in another PR (since we will add some tests for that, I guess we didnt have any test for task tracker because we dont get this error in our tests) if we want to since this PR is getting way too big

Reply:
maybe we should merge a PR that adds a test first.

Reply:
makes sense

Reply:
daed87c